### PR TITLE
feat(provider): respect contentful rate limit headers

### PR DESCRIPTION
**Pull request by cysp**

### CODE REVIEW

File: internal/provider/util/retry_backoff.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	resetSeconds, ok := parseContentfulRateLimitReset(resp.Header)
+	if !ok {
+		return retryablehttp.RateLimitLinearJitterBackoff(minDelay, maxDelay, attemptNum, resp)
+	}
+
+	return time.Duration(resetSeconds)*time.Second + contentfulRateLimitJitter()
+}
+
```
[the remark this answers is not in the document]
Comment (cysp) [reply internal/provider/util/retry_backoff.go]: Addressed locally: I now apply a `RetryWaitMin` floor for header-driven delays (`applyMinBackoffFloor`) so `X-Contentful-RateLimit-Reset: 0` no longer schedules sub-min retries. I also added a regression test for the zero-reset path.

File: internal/provider/util/retry_backoff.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+func contentfulRateLimitJitter() time.Duration {
+	jitterRangeMilliseconds := int64((rateLimitJitterMax-rateLimitJitterMin)/time.Millisecond) + 1
+
+	randomJitterMilliseconds, err := rand.Int(rand.Reader, big.NewInt(jitterRangeMilliseconds))
+	if err != nil {
+		return rateLimitJitterMin
+	}
+
+	return rateLimitJitterMin + time.Duration(randomJitterMilliseconds.Int64())*time.Millisecond
```
[the remark this answers is not in the document]
Comment (cysp) [reply internal/provider/util/retry_backoff.go:81]: Kept `crypto/rand` intentionally for this jitter helper. In this repo we avoid non-crypto PRNG in new backoff jitter paths to stay clear of gosec noise and keep behavior consistent with existing lint/security expectations; call frequency here is low and jitter range is small. No functional change needed from this comment.
